app/operations/users/destroy_self.rb

```diff
@@ -0,0 +1,24 @@
+# frozen_string_literal: true
+
+module Users
+  class DestroySelf < Actor
+    input :user, type: User
+    input :password, type: String
+
+    def call
+      return fail!(error: "Wrong password", status: :unauthorized) unless valid_password?
```

Comment:
Could you try without the return command if things work?
```suggestion
      fail!(error: "Wrong password", status: :unauthorized) unless valid_password?
```